doc/api/esm.md

```diff
@@ -191,6 +191,12 @@ also be used within `"exports"` to define different package entry points per
 environment, including whether the package is referenced via `require` or via
 `import`.
 
+**Warning**: Introducing the `"exports"` field prevents consumers of a package
+from using any entry points that are not defined, including the `package.json`.
+This will likely be a breaking change. To make the introduction of `"exports"`
+non-breaking, either ensure that every previously supported entry point is
+exported or include an export of the root of the package `"./": "./"`.
```

Comment:
```suggestion
This will likely be a breaking change. To make the introduction of `"exports"`
non-breaking, ensure that every previously supported entry point is exported.
It is best to explicitly specify entry points so that the package's public API is
well-defined. As a last resort, package encapsulation can be disabled entirely by
creating an export for the root of the package `"./": "./"` that will expose every file.
```

Reply:
This seems like a good start. Perhaps an example with `package.json` as well would make sense.